edi_oca/models/edi_configuration.py

```diff
@@ -0,0 +1,222 @@
+        It allows users to create their own configurations, which can be tailored
+        to meet the specific needs of their business processes.
+    """
+
+    name = fields.Char(string="Name", required=True)
+    active = fields.Boolean(default=True)
+    code = fields.Char(required=True, copy=False, index=True, unique=True)
+    description = fields.Char(help="Describe what the conf is for")
+    backend_id = fields.Many2one(string="Backend", comodel_name="edi.backend")
+    type_id = fields.Many2one(
+        string="Exchange Type",
+        comodel_name="edi.exchange.type",
+        ondelete="cascade",
+        auto_join=True,
+        index=True,
+    )
+    model = fields.Many2one(
+        "ir.model",
+        string="Model",
+        help="Model the conf applies to. Leave blank to apply for all models",
+    )
+    model_name = fields.Char(related="model.model", store=True)
+    partner_ids = fields.Many2many(
+        string="Enabled for partners",
+        comodel_name="res.partner",
+        relation="res_partner_edi_configuration_rel",
+        column1="conf_id",
+        column2="partner_id",
+        help="Leave blank to apply for all partners.",
+    )
+    trigger = fields.Selection(
+        [
+            ("on_record_write", "Update Record"),
+            ("on_record_create", "Create Record"),
+            ("on_email_send", "Send Email"),
+        ],
+        string="Trigger",
+        default=False,
+    )
+    snippet_before_do = fields.Text(
+        string="Snippet Before Do",
+        help="Snippet to validate the state and collect records to do",
+    )
+    snippet_do = fields.Text(
+        string="Snippet Do",
+        help="""Used to do something specific here.
+        Receives: operation, edi_action, vals, old_vals.""",
+    )
+
+    @api.constrains("backend_id", "type_id")
+    def _constrains_backend(self):
+        for rec in self:
+            if rec.type_id.backend_id:
+                if rec.type_id.backend_id != rec.backend_id:
+                    raise exceptions.ValidationError(
+                        _("Backend must match with exchange type's backend!")
+                    )
+            else:
+                if rec.type_id.backend_type_id != rec.backend_id.backend_type_id:
+                    raise exceptions.ValidationError(
+                        _("Backend type must match with exchange type's backend type!")
+                    )
+
+    @api.model
+    def default_get(self, fields):
+        vals = super().default_get(fields)
+        model = self.env.context.get("default_model_name", False)
+        partners = self.env.context.get("default_partner_ids", False)
+        if model:
+            vals["model"] = self.env["ir.model"]._get_id(model)
+        if partners and isinstance(partners, list):
+            vals["partner_ids"] = [(6, 0, partners)]
+        return vals
+
+    def _code_snippet_valued(self, snippet):
```

Comment:
this stuff is copied from edi_exchange_template and the type auto module. Can you please add a TODO to move it to some utils and/or mixin?